### ISSUE TITLE
yq 4.8 crashes when using assign operator on key that doesn't exist

### ISSUE THREAD

**Issue by sguillope** [mikefarah/yq#810]
**Describe the bug**
I'm attempting to remove keys from an object that don't match a specific value. The recent introduction of `with_entries` in 4.8.0 has solved this particular problem. However, it crashes when the object to be filter is not present. Using the optional identifier flag `?` doesn't help.
See discussion https://github.com/mikefarah/yq/discussions/809 for reference.

Version of yq: 4.8.0
Operating system: mac
Installed via: docker

**Input Yaml**
```yaml
some_key: some value
```

**Command**
The command you ran:
```
yq e '.environment |= with_entries(select(.key=="prod"))'
```

**Actual behavior**

It crashes if the key `environment` is not present.
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x5f20f7]

goroutine 1 [running]:
container/list.(*List).Front(...)
	/usr/local/go/src/container/list/list.go:70
github.com/mikefarah/yq/v4/pkg/yqlib.assignUpdateOperator(0x8fa788, 0xc00012f560, 0x0, 0x0, 0xc0001c0c60, 0x0, 0x0, 0x199, 0x101, 0xc0001cdb28)
	/go/src/mikefarah/yq/pkg/yqlib/operator_assign.go:25 +0x137
github.com/mikefarah/yq/v4/pkg/yqlib.(*dataTreeNavigator).GetMatchingNodes(0x8fa788, 0xc00012f560, 0x0, 0x0, 0xc0001c0c60, 0xc000000000, 0xc0000963d8, 0x0, 0x0, 0x203000)
	/go/src/mikefarah/yq/pkg/yqlib/data_tree_navigator.go:37 +0x2a5
github.com/mikefarah/yq/v4/pkg/yqlib.(*streamEvaluator).Evaluate(0xc00012f0e0, 0x7ffd7e187f7f, 0x9, 0x7514c0, 0xc0000963d8, 0xc0001c0c60, 0x7529e0, 0xc000098600, 0xc0000b6201, 0xc0001c0a60)
	/go/src/mikefarah/yq/pkg/yqlib/stream_evaluator.go:100 +0x368
github.com/mikefarah/yq/v4/pkg/yqlib.(*streamEvaluator).EvaluateFiles(0xc00012f0e0, 0x7ffd7e187f36, 0x48, 0xc0000a85d0, 0x1, 0x1, 0x7529e0, 0xc000098600, 0xc0000bdd80, 0xc0000bdd10)
	/go/src/mikefarah/yq/pkg/yqlib/stream_evaluator.go:63 +0x152
github.com/mikefarah/yq/v4/cmd.evaluateSequence(0xc0000c2c80, 0xc0000a85c0, 0x2, 0x2, 0x0, 0x0)
	/go/src/mikefarah/yq/cmd/evalute_sequence_command.go:115 +0x92a
github.com/spf13/cobra.(*Command).execute(0xc0000c2c80, 0xc0000a8560, 0x2, 0x2, 0xc0000c2c80, 0xc0000a8560)
	/go/src/mikefarah/yq/vendor/github.com/spf13/cobra/command.go:852 +0x47c
github.com/spf13/cobra.(*Command).ExecuteC(0xc0000c2a00, 0x0, 0x69d020, 0xc000080058)
	/go/src/mikefarah/yq/vendor/github.com/spf13/cobra/command.go:960 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/go/src/mikefarah/yq/vendor/github.com/spf13/cobra/command.go:897
main.main()
	/go/src/mikefarah/yq/yq.go:11 +0x2a
```

**Expected behavior**

```yaml
some_key: some value
```

**Additional context**
See discussion https://github.com/mikefarah/yq/discussions/809 for reference.

**Comment by mikefarah** [comment]
Thanks @sguillope - there is a work around available till I fix this bug:

```
yq e '(select(has("environment")) | .environment) |= with_entries(select(.key=="prod"))'
```

**Comment by mikefarah** [comment]
Fixed in 4.9.0
